client/apps/articles_list/components/listView.jsx

```diff
@@ -0,0 +1,147 @@
+  }
+
+  appendMore (results) {
+    const articles = this.state.articles.concat(results)
+    this.setState({articles})
+    return $('.loading-spinner').fadeOut()
+  }
+
+  fetchFeed (type, offset, cb) {
+    const feedQuery = query(`published: ${type}, offset: ${offset}, channel_id: "${sd.CURRENT_CHANNEL.id}"`)
+    return request
+      .post(sd.API_URL + '/graphql')
+      .set('X-Access-Token', sd.USER != null ? sd.USER.access_token : undefined)
+      .send({query: feedQuery})
+      .end((err, res) => {
+        if (err || !(res.body != null ? res.body.data : undefined)) { return }
+        this.setState({offset: this.state.offset + 10})
+        return cb(res.body.data.articles)
+      })
+  }
+
+  showEmptyMessage () {
+    return (
+      <div className='article=list__empty'>
+        <div>You haven’t written any articles yet.</div>
+        <div>Artsy Writer is a tool for writing stories about art on Artsy.</div>
+        <div>Get started by writing an article or reaching out to your liaison for help.</div>
+        <a
+          className='avant-garde-button avant-garde-button-black article-new-button'
+          href='/articles/new'
+          dangerouslySetInnerHTML={{ __html: $(icons()).filter('.new-article').html() + 'Write An Article' }}
+        />
+      </div>
+    )
+  }
+
+  showArticlesList () {
+    if (this.props.articles && this.props.articles.length) {
+      return (
+        <div className='articles-list__container'>
+          <div className='articles-list__title'>Latest Articles</div>
+          <FilterSearch
+            url={sd.API_URL + `/articles?published=${this.state.published}&channel_id=${sd.CURRENT_CHANNEL.id}&q=%QUERY`}
+            placeholder='Search Articles...'
+            collection={this.state.articles}
+            searchResults={this.setResults}
+            selected={null}
+            contentType='article'
+            checkable={false}
+            isArtsyChannel={this.props.channel.isArtsyChannel()}
+          />
+        </div>
+      )
+    } else {
+      return this.showEmptyMessage()
+    }
+  }
+
+  render () {
+    return (
+      <div className='articles-list'>
+        <h1 className='articles-list__header page-header'>
+          <div className='max-width-container'>
+            <nav className='nav-tabs'>
+              <a className={`${this.state.published === true ? 'is-active' : ''} published`}
+                onClick={() => this.setPublished(true)}>
+                Published
+              </a>
+              <a className={`${this.state.published === false ? 'is-active' : ''} drafts`}
+                onClick={() => this.setPublished(false)}>
+                Drafts
+              </a>
+            </nav>
+            <div className='channel-name'>
+              {`${this.props.channel.get('name')}`}
+            </div>
+          </div>
+        </h1>
+        {this.showArticlesList()}
+      </div>
+    )
+  }
+}
+
+const mapStateToProps = (state) => ({
+  ...state
+})
```

Comment:
Can do `mapStateToProps = state => state`

Reply:
Ideally though, rather than returning the entire state it's best to just pluck what you need, so that the API boundaries of the component are clearly defined:

```javascript
const mapStateToProps = (state) => ({
  foo: state.foo,
  baz: stare.bar.baz
})
```

Reply:
Yes, definitely. We weren't using the redux state in this component which is why is why I didn't explicitly list out any props.

Reply:
If not using can probably remove for now